rust/src/shards/gui/widgets/hex_viewer.rs

```diff
@@ -0,0 +1,192 @@
+
+    Some(&self.requiring)
+  }
+
+  fn warmup(&mut self, context: &Context) -> Result<(), &str> {
+    self.parents.warmup(context);
+
+    Ok(())
+  }
+
+  fn cleanup(&mut self) -> Result<(), &str> {
+    self.parents.cleanup();
+
+    Ok(())
+  }
+
+  fn activate(&mut self, _context: &Context, input: &Var) -> Result<Var, &str> {
+    use egui_memory_editor::MemoryEditor;
+
+    if let Some(ui) = util::get_current_parent(*self.parents.get())? {
+      let mem = unsafe {
+        let (data, len) = match input.valueType {
+          SHType_Bytes => (
+            input.payload.__bindgen_anon_1.__bindgen_anon_4.bytesValue,
+            input.payload.__bindgen_anon_1.__bindgen_anon_4.bytesSize as usize,
+          ),
+          SHType_Enum => (
+            &input.payload.__bindgen_anon_1.__bindgen_anon_3.enumValue as *const i32 as *mut u8,
+            4,
+          ),
+          SHType_Float => (
+            &input.payload.__bindgen_anon_1.floatValue as *const f64 as *mut u8,
+            8,
+          ),
+          SHType_Float2 => (
+            &input.payload.__bindgen_anon_1.float2Value as *const f64 as *mut u8,
+            16,
+          ),
+          SHType_Float3 => (
+            &input.payload.__bindgen_anon_1.float3Value as *const f32 as *mut u8,
+            12,
+          ),
+          SHType_Float4 => (
+            &input.payload.__bindgen_anon_1.float4Value as *const f32 as *mut u8,
+            16,
+          ),
+          SHType_Int => (
+            &input.payload.__bindgen_anon_1.intValue as *const i64 as *mut u8,
+            8,
+          ),
+          SHType_Int2 => (
+            &input.payload.__bindgen_anon_1.int2Value as *const i64 as *mut u8,
+            16,
+          ),
+          SHType_Int3 => (
+            &input.payload.__bindgen_anon_1.int3Value as *const i32 as *mut u8,
+            12,
+          ),
+          SHType_Int4 => (
+            &input.payload.__bindgen_anon_1.int4Value as *const i32 as *mut u8,
+            16,
+          ),
+          SHType_Int8 => (
+            &input.payload.__bindgen_anon_1.int8Value as *const i16 as *mut u8,
+            16,
+          ),
+          SHType_Int16 => (
+            &input.payload.__bindgen_anon_1.int16Value as *const i8 as *mut u8,
+            16,
+          ),
+          SHType_String => (
+            input.payload.__bindgen_anon_1.__bindgen_anon_2.stringValue as *mut u8,
+            input.payload.__bindgen_anon_1.__bindgen_anon_2.stringLen as usize,
+          ),
+          _ => unreachable!(),
+        };
+        std::slice::from_raw_parts_mut(data, len)
+      };
+
+      let mut mem_editor = MemoryEditor::new().with_address_range("All", 0..mem.len());
+
+      if let Some(options) = &self.editor_options {
+        mem_editor = mem_editor.with_options(options.clone());
+      }
+      mem_editor.draw_editor_contents_read_only(ui, mem, |mem, address| mem[address].into());
+      self.editor_options = Some(mem_editor.options.clone());
```

Comment:
I think you can keep the MemoryEditor instance on this shard, that way you also don't have to get/restore the options every frame.

Reply:
Just update the address range potentially

Reply:
yeah.... I think that needs to be done

Reply:
Unfortunately it can't be done. The address range is not writeable, except during construction.

Reply:
I mean all the `new` around egui

Reply:
I think we discussed this on discord